cmd/rofl/deploy.go

```diff
@@ -249,6 +251,29 @@ var (
 	}
 )
 
+// detectTerm returns the preferred (longest) period of the given offer.
+func detectTerm(offer *roflmarket.Offer) (term roflmarket.Term) {
+	if offer == nil || offer.Payment.Native == nil {
+		cobra.CheckErr(fmt.Errorf("unable do determine payment term for offer '%s'", offer.ID))
+	}
+
+	if deployTerm != "" {
+		// Custom deploy term.
+		term = roflCommon.ParseMachineTerm(deployTerm)
+		if _, ok := offer.Payment.Native.Terms[term]; !ok {
+			cobra.CheckErr(fmt.Errorf("term '%s' is not available for offer '%s'", deployTerm, offer.ID))
+		}
+		return
+	}
+
+	for t := range offer.Payment.Native.Terms {
+		if t > term {
```

Comment:
This is comparing the internal term identifier? What if we add `TermDay` with id `4`?

Reply:
This would require Oasis SDK changes. I'll add a remark for now.